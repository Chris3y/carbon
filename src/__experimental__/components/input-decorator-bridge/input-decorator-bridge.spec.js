import React from 'react';
import { shallow } from 'enzyme';
import InputDecoratorBridge from './input-decorator-bridge.component';

jest.mock('../../../utils/helpers/guid', () => () => 'mocked-guid');

describe('InputDecoratorBridge', () => {
  const shallowRender = props => (
    shallow(
      <InputDecoratorBridge
        inputRef={ () => {} }
        value='1'
        { ...props }
      />
    )
  );

  it(`renders the Textbox component with:
         * generated props from the decorators assigned to correct elements
         * markup for label, validation and field help
         * children and left children`, () => {
    const wrapper = shallowRender({
      fieldHelp: 'please help!',
      children: 'normal children',
      leftChildren: 'southpaw children'
    });
    wrapper.setState({ valid: false });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with an inputIcon if one is supplied', () => {
    const wrapper = shallowRender({ inputIcon: 'dropdown' });
    expect(wrapper).toMatchSnapshot();
  });

  it('uses a formatted value on the Textbox if one is supplied', () => {
    const wrapper = shallowRender({ formattedValue: 'formatted!' });
    expect(wrapper.find('Textbox').props().value).toEqual('formatted!');
  });

  it('assigns custom html attributes to Textbox', () => {
    const wrapper = shallowRender({ placeholder: 'placeholder value' });
    expect(wrapper.find('Textbox').props().placeholder).toEqual('placeholder value');
  });
});
