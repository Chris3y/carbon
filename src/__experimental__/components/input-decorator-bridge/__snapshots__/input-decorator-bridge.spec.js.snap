// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDecoratorBridge renders the Textbox component with:
         * generated props from the decorators assigned to correct elements
         * markup for label, validation and field help 1`] = `
<div
  className="common-input--with-icon common-input--error common-input--has-field-help common-input"
>
  <div
    className="common-input__field"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <Textbox
      autoComplete="off"
      data-element="input"
      id="mocked-guid"
      inputRef={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value="1"
    />
  </div>
  <Icon
    bgSize="small"
    className="common-input__icon common-input__icon--error"
    key="0"
    style={Object {}}
    type="error"
  />
  <span
    className="common-input__help-text"
    data-element="help"
    style={Object {}}
  >
    please help!
  </span>
</div>
`;

exports[`InputDecoratorBridge renders with an inputIcon if one is supplied 1`] = `
<div
  className="common-input--with-icon common-input"
>
  <div
    className="common-input__field"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <Textbox
      autoComplete="off"
      data-element="input"
      id="mocked-guid"
      inputRef={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value="1"
    />
    <label
      htmlFor="mocked-guid"
      key="label-icon"
    >
      <Icon
        bgSize="small"
        className="carbon-input-icon"
        type="dropdown"
      />
    </label>
  </div>
</div>
`;
